### DISCUSSION TITLE
[Question] Ready to replace Flink SQL?

### DISCUSSION THREAD

**Discussion by zhanglistar** [ArroyoSystems/arroyo#644] [General]
Very glad to see arroyo like project that using native language to rewrite and redesign streaming computation engine. 
We are finding high performance streaming computation engine and we can contribute code to such project.
Please tell me what's the progress of feature to replace Flink SQL. 
Thanks a lot.

**Comment by mwylde** [comment]
Thanks for checking out Arroyo! Arroyo is able to replace Flink SQL for many use cases. Compared to Flink, arroyo is much faster (5-10x depending on the query), easier to develop against (better error messages, better UI, some simpler SQL semantics) and easier to run (no JVM, designed for fast rescaling/recovery and to run in the cloud and kubernetes).

However, today Flink SQL has more features (for example, we don't currently support all of the [joins](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/sql/queries/joins/) that Flink does, more connectors, and a much bigger community. I'd take a look our [SQL docs](https://doc.arroyo.dev/sql/data-types) and see if the features you need are there. Feel free to open up issues for any features that you need and we can see how difficult they would be and where they fit into our prioritization.

**Reply by zhanglistar** [reply]
Thanks for your reply. I will dive into it.
